readthedocs/redirects/templates/redirects/redirect_form.html

```diff
@@ -39,31 +57,11 @@
         };
 
         $(document).ready(function() {
-            $('#id_from_url').parent().hide()
-            $('#id_to_url').parent().hide()
-            $('#dynamic-redirect').hide()
-
+            update_outcome();
             $('#id_redirect_type').bind('change', function(ev) {
-                $('#id_from_url').val('');
-                $('#id_to_url').val('');
-                $('#id_from_url').parent().hide()
-                $('#id_to_url').parent().hide()
-
-                if (this.value == 'prefix') {
-                    $('#id_from_url').parent().show()
-                    $('#id_to_url').parent().hide()
-                }
-                else if (this.value == 'page') {
-                    $('#id_from_url').parent().show()
-                    $('#id_to_url').parent().show()
-                }
-                else if (this.value == 'exact') {
-                    $('#id_from_url').parent().show()
-                    $('#id_to_url').parent().show()
-                }
```

Comment:
moved this logic to be inside the `update_outcome` function, this way it works when the form has pre-populated fields when editing an existing redirect.

Reply:
I didn't fully check this Javascript code, but looks good!